Sources/MongoSwift/MongoCollection+FindAndModify.swift

```diff
@@ -0,0 +1,127 @@
+import libmongoc
+
+/// An extension of `MongoCollection` encapsulating find and modify operations.
+extension MongoCollection {
+    /**
+     * Finds a single document and deletes it, returning the original.
+     *
+     * - Parameters:
+     *   - filter: `Document` representing the match criteria
+     *   - options: Optional `FindOneAndDeleteOptions` to use when executing the command
+     *
+     * - Returns: The deleted document, or `nil` if no document was deleted.
+     */
+    @discardableResult
```

Comment:
I considered that discarding the result doesn't make sense, as users should probably use the single-statement methods (e.g. `deleteOne`); however, the fact that `findAndModify` supports a `sort` option justifies why someone may want to use these methods while still not caring about the return value. :+1: